languages/thingtalk/dialogue_acts/action-confirm.js

```diff
@@ -68,7 +72,37 @@ function actionConfirmAcceptPhrase(ctx) {
     return clone.state;
 }
 
+function actionConfirmRejectPhrase(ctx) {
+    return new Ast.DialogueState(null, POLICY_NAME, 'cancel', null, []);
+}
+
+function actionConfirmChangeParam(ctx, answer) {
+    const questions = ctx.dialogueActParam || [];
+    if (answer instanceof Ast.Value) {
+        if (questions.length !== 1)
+            return null;
+        answer = new Ast.InputParam(null, questions[0], answer);
+    }
+
+    const action = C.getInvocation(ctx.state.history[0]);
+    if (!action) return null;
+
+    // don't continue with queries
+    if (action.schema._functionType === "query") return null;
+
+    // don't accept in params that don't apply to this specific action
+    const allowedParams = new Set(Object.keys(ctx.nextFunctionSchema._inReq));
+    allowedParams.add(Object.keys(ctx.nextFunctionSchema._inOpt));
+    if (!allowedParams.has(answer.name)) return null;
```

Comment:
It's easier to write this as
```
const arg = ctx.nextFunctionSchema.getArgument(answer.name);
if (!arg || !arg.is_input)
```
You also need to check that the type is correct.